app/components/Dropdown/index.js

```diff
@@ -11,82 +11,69 @@ import styles from './Dropdown.css';
 type Props = {
   iconName?: string,
   toggle: () => any,
+  closeOnClick?: boolean,
   className?: string,
   contentClassName?: string,
   componentClass?: any,
   triggerComponent?: Node | Portal,
   show: boolean,
   children?: any,
   style?: any,
-  placement?: 'top' | 'bottom' | 'left' | 'right',
 };
 
-class Dropdown extends Component<Props> {
-  target: any;
+const Dropdown = ({
+  iconName = 'star',
+  toggle,
+  closeOnClick = false,
+  className,
+  contentClassName,
+  componentClass: ComponentClass = 'button',
+  triggerComponent,
+  show,
+  children,
+  style,
+}: Props) => {
+  const triggerRef = useRef(null);
 
-  static ListItem: (any) => Node = ListItem;
-  static List: ({ children: any }) => Node = List;
-  static Divider: () => Node = Divider;
+  return (
+    <ComponentClass
+      onClick={show ? null : toggle} // avoid double toggle because of rootClose
+      ref={triggerRef}
+      className={className}
+      style={style}
+    >
+      {triggerComponent || (iconName ? <Icon name={iconName} /> : null)}
 
-  renderContent(): Node {
-    if (this.props.triggerComponent) {
-      return this.props.triggerComponent;
-    }
-
-    const { iconName = 'star' } = this.props;
-    return iconName ? <Icon name={iconName} /> : null;
-  }
-
-  render(): Node {
-    const {
-      toggle,
-      show,
-      contentClassName,
-      className,
-      children,
-      style,
-      placement = 'bottom',
-      componentClass: ComponentClass = 'button',
-    } = this.props;
-
-    return (
-      <ComponentClass
-        onClick={toggle}
-        ref={(target) => {
-          this.target = target;
-        }}
-        className={className}
-        style={style}
+      <Overlay
+        show={show}
+        onHide={toggle}
+        target={triggerRef}
+        rootClose
+        shouldUpdatePosition
       >
-        {this.renderContent()}
-
-        <Overlay
-          show={show}
-          onHide={toggle}
-          target={this.target}
-          placement={placement}
-          rootClose
-          shouldUpdatePosition
-        >
-          <div className={cx(styles.content, contentClassName || null)}>
+        {({ props, arrowProps }) => (
+          <div
+            {...props}
+            className={cx(styles.content, contentClassName || null)}
+            onClick={closeOnClick && toggle}
```

Comment:
Is this the same as `rootClose` in the old version? If so, could you call it something else. Having it be `closeOnClick` onthe Dropdown is kinda counterintuitive and I think `closeOnClickOutside` or similar is more intuitive.

Or `closeOnBackdropClick` as I see it's called in the `Modal`.

Reply:
This actually toggles closing the dropdown when clicking the dropdown content. After I updated it f.ex. the user dropdown in the header would stay open after clicking settings and going to the settings page. I will rename it to `closeOnContentClick` to make it clearer.

Reply:
I personally thought `closeOnClick` made sense, and it also sort of "matches" `ConfirmModal`'s `closeOnConfirm` prop.

Reply:
Ah okay then it makes sense yeah, I though it was "close on click outside the dropdown" :sweat_smile:.
But yeah I think `closeOnContentClick` is just a bit clearer :+1:

Reply:
Yup, I like `closeOnContentClick`. I don't think it hurts to be a little verbose